test/prob/discrete_range/discrete_range_ccdf_log_test.hpp

```diff
@@ -0,0 +1,65 @@
+// Arguments: Ints, Ints, Ints
+#include <stan/math/prim.hpp>
+
+using stan::math::var;
+using std::vector;
+
+class AgradCcdfLogDiscreteRange : public AgradCcdfLogTest {
+ public:
+  void valid_values(vector<vector<double>>& parameters,
+                    vector<double>& ccdf_log) {
+    vector<double> param(3);
+
+    param[0] = 3;  // y
+    param[1] = 1;  // lower
+    param[2] = 5;  // upper
+    parameters.push_back(param);
+    ccdf_log.push_back(-0.916290731874154995573);  // expected ccdf_log
+
+    param[0] = 9;   // y
+    param[1] = 5;   // lower
+    param[2] = 15;  // upper
+    parameters.push_back(param);
+    ccdf_log.push_back(-0.606135803570315601085);  // expected ccdf_log
+
+    param[0] = 0;   // y
+    param[1] = -4;  // lower
+    param[2] = 5;   // upper
+    parameters.push_back(param);
+    ccdf_log.push_back(-0.693147180559945286227);  // expected ccdf_log
```

Comment:
The upper boundary should be tested for completeness, y = upper.  There the value should be `LOG_ZERO`. It's in support, but has a zero CCDF.

Reply:
I've added prim tests for boundaries. I'm struggling with the distribution test framework to behave as desired for boundaries/exceptions, it just doesn't seem designed to support those. I'll try to summarise my findings about that when I've got a better handle on thosse.